website/docs/vision.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Vision
+sidebar_label: Vision
+sidebar_position: 2
+---
+
+## Charter
+
+**One Layer, All Storage.**
+
+## Principles
+
+Tenets are guiding principles. They guide how decisions are made for the whole project. Ideally, we do all of them all the time. In some cases, though, we may be forced to decide between slightly penalizing one goal or another. In that case, we tend to support those goals that come earlier in the list over those that come later (but every case is different).
+
+### 0. Open Community
+
+OpenDAL SHOULD be an **open** storage library.
+
+OpenDAL is an ASF project governed by the OpenDAL PMC. At ASF, we believe in "Community Over Code" and adhere to [the Apache Way](https://www.apache.org/theapacheway/). We aim to develop OpenDAL to meet the needs of our community. We do not maintain private versions or include features that aren't useful to others.
+
+For example, OpenDAL prefers to have clear and readable code, as this allows more people in the community to join the development.
+
+### 1. Solid Foundation
```

Comment:
-1 for `Solid Foundation`, as mentioned in https://github.com/apache/opendal/discussions/5301#discussioncomment-11220635

IMO such a tenet is too big, hollow and subjective to provide guidance meaning and be constructive: no one will ever want to uncarefully build unsolid, untrustworthy software. It is self-evident to me that don't need to be told.

I propose the following alternatives:
- `Well-defined Behavior`
- Or Simply remove this one

Reply:
> no one will ever want to uncarefully build unsolid, untrustworthy software. It is self-evident to me that don't need to be told.

This is true. The closest comparison I can think of is the set of safety rules for flying a plane. No one wants the plane to crash, be damaged, or get lost, but that does happen. This principle exists to guide our development so that we don't make changes that lead us off track.

> IMO such a tenet is too big, hollow and subjective to provide guidance meaning and be constructive

Yes, I agree that "solid foundation" is somewhat a big word. I'm open to better alternatives.

> `Well-defined Behavior`

This phrase doesn't convey what we want in these areas.

> Or Simply remove this one

Unfortunately, I believe we need to emphasize this area. Removing this one might leave our principles incomplete. I still suggest to use `solid foundation` if we don't have better one.